Updated CoreObject Examples

TypicalHB opened the pull request:
Updated some CoreObject examples.

CoreObject.childAddedEvent
CoreObject.childRemovedEvent
CoreObject.descendantAddedEvent
CoreObject.descendantRemovedEvent

TypicalHB committed:
Updated CoreObject Examples

Updated some more examples. All CoreObject examples.

CoreObject.childAddedEvent
CoreObject.childRemovedEvent
CoreObject.descendantAddedEvent
CoreObject.descendantRemovedEvent

Montoli commented on src/api/examples.md:
```diff
@@ -1484,48 +1484,50 @@ AssignMeshToTeam(cube, 1)
-function OnChildAdded()
-    print("A child has been added to template 1!")
-end
+obj.descendantAddedEvent:Connect(function() -- This will run when a child gets added to the object.
```
```suggestion
obj.descendantAddedEvent:Connect(function() -- This will run when a descendant gets added to the object.
```

Montoli commented on src/api/examples.md:
````diff
@@ -1484,48 +1484,50 @@ AssignMeshToTeam(cube, 1)
 
-### <a id="event:CoreObject.descendantRemovedEvent"></a>CoreObject.descendantRemovedEvent
+```lua
+local obj = script.parent -- The object that will be affected by this event.
````
We've been trying to avoid using `script.parent` as a core object in these samples, for a couple of reasons.

First, because we don't know what the hierarchy looks like, and we want to avoid potential side effects if people just plunk the examples into the game.  (Admittedly side effects from attaching an event listener are minor compared to, say, an example that called `script.parent:Destroy()` or something.  But we like them to be consistent.)

Second though, these scripts also get run by an automated process that verifies that all the examples work before publishing.  And due to how it is set up, `script.parent` is not guaranteed to be a good thing to use in samples.

Anyway!  TL;DR:
If a sample needs a `CoreObject` to operate on, we'd prefer if the code creates it itself (via something like `local template1 = World.SpawnAsset(propCubeTemplate)` than use `script.parent`.

Montoli reviewed (changes requested):
Hey!  Thanks for helping us make the Core documentation better!

I see two minor problems - See the comment about `script.parent`, and fix the comments, and we'll fold these in to the official docs.

Thank you!

TypicalHB committed:
Update examples.md

TypicalHB commented:
> Hey! Thanks for helping us make the Core documentation better!
> 
> I see two minor problems - See the comment about `script.parent`, and fix the comments, and we'll fold these in to the official docs.
> 
> Thank you!

Fixed. Everything should be good now. I also updated my previous "CoreObject:Destroy()". Everything should be fine now. :)

nicholasforeman reviewed (commented):
Really good additions, especially with the new snippet for childAddedEvent.

Also, I like the encouragement of custom properties instead of script.parent, however for many beginners it is hard to track custom properties.

Edit: Just saw the comment by Montoli about script.parent, didn't realize that's what they are trying to get away from. Makes sense as it's a better convention and it's best to encourage that from the beginning.
